Missing dependencies in Singularity recipe file

Building the Singularity container image does not result in a working container.  The build of Singularity fails due to missing dependencies.

```
$ sudo /usr/local/bin/singularity build sregistry-cli Singularity
... 
+ ./configure --prefix=/usr/local
...
checking for python... no
configure: error: Please install python before installing.
...
checking archive.h usability... no
checking archive.h presence... no
checking for archive.h... no
configure: error: Unable to find the libarchive headers, need package libarchive-devel (libarchive-dev on Debian/Ubuntu)
```

This results in a non-working container:

```
$ ./sregistry-cli pull ...
WARNING Singularity is not installed, function might be limited.
WARNING Singularity is not installed, function might be limited.
...
ERROR Cannot find Singularity! Is it installed?
```

I was able to work around the issue by adding `python` and `libarchive-dev` to the list of packages installed by `apt-get`.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ and **Merged pull requests**. Critical items to know are:
 The versions coincide with releases on pip. Only major versions will be released as tags on Github.
 
 ## [0.0.x](https://github.com/singularityhub/sregistry-cli/tree/master) (0.0.x)
+ - updating Dockerfile and Singularity recipt with additional dependencies for 2.5 (0.0.86).
  - removing Globus dependency to avoid install conflict (0.0.85)
  - added client command to add, list, activate, and remove backends (0.0.84)
  - bumping spython dependency to 0.0.25 (0.0.83)

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,14 +7,6 @@ FROM continuumio/miniconda3
 # docker run vanessa/sregistry-cli
 #########################################
 
-RUN mkdir /code
-ADD . /code
-RUN /opt/conda/bin/pip install setuptools
-RUN /opt/conda/bin/pip install scif
-
-RUN scif install /code/sregistry-cli.scif
-ENTRYPOINT ["scif"]
-
 ENV PATH /usr/local/bin:$PATH
 LABEL maintainer vsochat@stanford.edu
 
@@ -36,6 +28,14 @@ RUN git clone https://www.github.com/singularityware/singularity.git
 WORKDIR singularity
 RUN ./autogen.sh && ./configure --prefix=/usr/local && make && make install
 
+RUN mkdir /code
+ADD . /code
+RUN /opt/conda/bin/pip install setuptools
+RUN /opt/conda/bin/pip install scif
+
+RUN scif install /code/sregistry-cli.scif
+ENTRYPOINT ["sregistry"]
+
 WORKDIR /code
 
-RUN /opt/conda/bin/pip install -e .
+RUN /opt/conda/bin/pip install -e .[all]

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,7 +7,6 @@ FROM continuumio/miniconda3
 # docker run vanessa/sregistry-cli
 #########################################
 
-ENV PATH /opt/conda/bin:/usr/local/bin:$PATH
 LABEL maintainer vsochat@stanford.edu
 
 RUN apt-get update && apt-get install -y git build-essential \
@@ -37,5 +36,6 @@ RUN scif install /code/sregistry-cli.scif
 ENTRYPOINT ["sregistry"]
 
 WORKDIR /code
+RUN rm /usr/bin/python && ln -s /opt/conda/bin/python /usr/bin/python
 
 RUN /opt/conda/bin/pip install -e .[all]

--- a/Dockerfile
+++ b/Dockerfile
@@ -36,6 +36,6 @@ RUN scif install /code/sregistry-cli.scif
 ENTRYPOINT ["sregistry"]
 
 WORKDIR /code
-RUN rm /usr/bin/python && ln -s /opt/conda/bin/python /usr/bin/python
+RUN rm /usr/bin/python && ln -sf /opt/conda/bin/python /usr/bin/python
 
 RUN /opt/conda/bin/pip install -e .[all]

--- a/Dockerfile
+++ b/Dockerfile
@@ -7,7 +7,7 @@ FROM continuumio/miniconda3
 # docker run vanessa/sregistry-cli
 #########################################
 
-ENV PATH /usr/local/bin:$PATH
+ENV PATH /opt/conda/bin:/usr/local/bin:$PATH
 LABEL maintainer vsochat@stanford.edu
 
 RUN apt-get update && apt-get install -y git build-essential \

--- a/Singularity
+++ b/Singularity
@@ -33,6 +33,16 @@ From: continuumio/miniconda3
     exec /opt/conda/bin/sregistry "$@"
 
 
+#######################################
+# Globus
+#######################################
+
+%appenv globus
+    SREGISTRY_CLIENT=globus
+    export SREGISTRY_CLIENT
+%apprun globus
+    exec /opt/conda/bin/sregistry "$@"
+
 
 #######################################
 # Singularity Hub
@@ -57,7 +67,7 @@ From: continuumio/miniconda3
 
 
 %environment
-    PATH=/usr/local/bin:$PATH
+    PATH=/usr/local/bin:/opt/conda/bin:$PATH
     export PATH
 
 
@@ -68,12 +78,14 @@ From: continuumio/miniconda3
     apt-get update && apt-get install -y git build-essential \
                    libtool \
                    squashfs-tools \
+                   libarchive-dev  \
                    autotools-dev \
                    automake \
                    autoconf \
                    uuid-dev \
                    libssl-dev
 
+    export PATH=/opt/conda/bin:$PATH
     /opt/conda/bin/pip install dateutils
 
     # Install Singularity
@@ -86,7 +98,4 @@ From: continuumio/miniconda3
     /opt/conda/bin/pip install setuptools
 
     # This installs all "install extras"
-    /opt/conda/bin/pip install -e .
-    /opt/conda/bin/pip install -e .[registry]
-    /opt/conda/bin/pip install -e .[google-drive]
-    /opt/conda/bin/pip install -e .[google-storage]
+    /opt/conda/bin/pip install -e .[all]

--- a/docs/_includes/editable.html
+++ b/docs/_includes/editable.html
@@ -43,7 +43,7 @@ $(document).ready(function() {
         var start = '<div class="dropdown more"><span><i class="fa fa-ellipsis-h more" title="Edit"></i></span><div class="dropdown-content">';
 
         // Permalink
-        var link = "{{ site.url }}/{{ site.baseurl }}/{{ page.url }}#" + did;
+        var link = "{{ site.url }}{{ site.baseurl }}/{{ page.url }}#" + did;
         var button = "<p><a href='" + link + "' target='_blank'>Permalink</a></p>";
         start += button;
 

--- a/docs/_includes/editable.html
+++ b/docs/_includes/editable.html
@@ -43,7 +43,7 @@ $(document).ready(function() {
         var start = '<div class="dropdown more"><span><i class="fa fa-ellipsis-h more" title="Edit"></i></span><div class="dropdown-content">';
 
         // Permalink
-        var link = "{{ site.url }}/{{ page.url }}#" + did;
+        var link = "{{ site.url }}/{{ site.baseurl }}/{{ page.url }}#" + did;
         var button = "<p><a href='" + link + "' target='_blank'>Permalink</a></p>";
         start += button;
 

--- a/docs/_includes/editable.html
+++ b/docs/_includes/editable.html
@@ -43,7 +43,7 @@ $(document).ready(function() {
         var start = '<div class="dropdown more"><span><i class="fa fa-ellipsis-h more" title="Edit"></i></span><div class="dropdown-content">';
 
         // Permalink
-        var link = "{{ site.url }}{{ site.baseurl }}/{{ page.url }}#" + did;
+        var link = "{{ site.url }}{{ site.baseurl }}{{ page.url }}#" + did;
         var button = "<p><a href='" + link + "' target='_blank'>Permalink</a></p>";
         start += button;
 

--- a/docs/install.md
+++ b/docs/install.md
@@ -6,20 +6,12 @@ permalink: /install
 toc: false
 ---
 
-# Installation Local
+# Local Install
 
 To install from the Github repository:
 
 ```bash
 git clone https://www.github.com/singularityhub/sregistry-cli.git
-cd sregistry-cli
-python setup.py install
-```
-
-We recommend for the latest features to use the development branch:
-
-```bash
-git clone -b development https://www.github.com/singularityhub/sregistry-cli.git
 cd sregistry-cli
 python setup.py install
 ```
@@ -56,17 +48,69 @@ or for a particular extra client:
 pip install sregistry[myclient]
 ```
 
-For now, it's probably fastest and easiest to use the Singularity image.
+# Docker
+You can use Singularity Registry Global clients via Docker. Importantly, you need to
+run the commands with the `--privileged` flag.
+
+```bash
+$ docker run --privileged vanessa/sregistry-cli pull shub://vsoch/hello-world
+```
+
+If you need to build the container locally:
+
+```bash
+$ docker build -t vanessa/sregistry-cli .
+```
+
+# Singularity
+Singularity no longer supports running Singularity inside Singularity, and so some commands
+will not work as desired. For example:
+
+```bash
+ ./sregistry.simg pull shub://vsoch/hello-world
+[client|hub] [database|sqlite:////home/vanessa/.singularity/sregistry.db]
+Progress |===================================| 100.0% 
+2.5.1-master.gd6e81547
+ERROR  : Singularity is not running with appropriate privileges!
+ERROR  : Check installation path is not mounted with 'nosuid', and/or consult manual.
+ABORT  : Retval = 255
+
+ERROR Return Code 255: ("\x1b[91mERROR  : Singularity is not running with appropriate privileges!\n\x1b[0m\x1b[91mERROR  : Check installation path is not mounted with 'nosuid', and/or consult manual.\n\x1b[0m\x1b[31mABORT  : Retval = 255\n\x1b[0m",)
+[container][update] vsoch/hello-world:latest@ed9755a0871f04db3e14971bec56a33f
+Success! /home/vanessa/.singularity/shub/vsoch-hello-world:latest@ed9755a0871f04db3e14971bec56a33f.simg
+```
+
+You could run with sudo, but that might defeat the purpose. Most functions should work, and
+missing is adding the inspection of the container to the database. It can, of course,
+easily be obtained by inspecting the container image directly.
+
+To activate a particular client endpoint, thanks to the [Standard Container Integration Format](https://containersftw.github.io/SCI-F/)
+you can just use an `--app` flag instead:
+
+```bash
+$ singularity run --app registry sregistry-cli
+```
 
 # Clients Available
 Singularity Registry Global Client is developed to give you maximum flexibility to install only what you need. For this, you have many different options for installing the software. if you use the above method, you will install the full client and storage, meaning that you can push and pull images to many different clients (Google vs. Singularity Registry) and you can keep a local database file for keeping track of your images. However, it might be that you want to install dependencies for just one client. Remember for all of the pip commands below, you can install with a local repository, or remote.
+
+Install all clients
+
+```bash
+# Local repository
+pip install -e .[all]
+
+# From pypi
+pip install sregistry[all]
+```
+
 
 ```bash
 # Local repository
 pip install -e .[myclient]
 
 # From pypi
-pip install [myclient]
+pip install sregistry[myclient]
 ```
 
 And here are your options.
@@ -100,24 +144,3 @@ pip install sregistry[google-drive-basic]
 
 Clients for Singularity Hub and Nvidia are not detailed here as they don't require additional library dependencies.
 
-
-# Singularity
-To build a singularity container
-
-```bash
-sudo singularity build sregistry-cli Singularity
-```
-
-And now anywhere in these pages where you run an sregistry command, instead just
-reference the image:
-
-```bash
-./sregistry-cli
-```
-
-and to activate a particular client endpoint, thanks to the [Standard Container Integration Format](https://containersftw.github.io/SCI-F/)
-you can just use an `--app` flag instead:
-
-```bash
-singularity run --app registry sregistry-cli
-```

--- a/sregistry/main/base/inspect.py
+++ b/sregistry/main/base/inspect.py
@@ -67,6 +67,7 @@ def get_metadata(self, image_file, names={}):
 
         # We try and inspect, but not required (wont work within Docker)
         try:
+            Singularity.quiet = True
             updates = Singularity.inspect(image=image_file)
         except:
             bot.warning('Inspect command not supported, metadata not included.')

--- a/sregistry/version.py
+++ b/sregistry/version.py
@@ -33,7 +33,7 @@ LICENSE = "LICENSE"
 
 
 INSTALL_REQUIRES = (
-    ('spython', {'min_version': '0.0.25'}),
+    ('spython', {'min_version': '0.0.34'}),
     ('requests', {'min_version': '2.18.4'}),
     ('pygments', {'min_version': '2.1.3'}),
 )

--- a/sregistry/version.py
+++ b/sregistry/version.py
@@ -19,7 +19,7 @@ along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 '''
 
-__version__ = "0.0.85"
+__version__ = "0.0.86"
 AUTHOR = 'Vanessa Sochat'
 AUTHOR_EMAIL = 'vsochat@stanford.edu'
 NAME = 'sregistry'

